plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServerSettingsWizardPage.java

```diff
@@ -1088,8 +1101,27 @@ public boolean isPageComplete() {
 
 	public IServer saveServer(IProgressMonitor monitor) throws CoreException {
 		model.updateServer();
+		setOpenshiftMavenProfileActive(model.getDeployProject());
 		return model.saveServer(monitor);
 	}
+	
+	@SuppressWarnings("restriction")
+    private void setOpenshiftMavenProfileActive(IProject newDeployProject)  throws CoreException  {
+	    IFile pom = newDeployProject.getFile(IMavenConstants.POM_FILE_NAME);
```

Comment:
the file may only be retrieved if the project has the maven nature, so the call could be easily moved to l1112 making it more logical, I think.